libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/ParameterizedBotFrameworkAuthentication.java

```diff
@@ -123,31 +125,32 @@ public String getOriginatingAudience() {
     @Override
     public CompletableFuture<AuthenticateRequestResult> authenticateStreamingRequest(String authHeader,
                                                                                      String channelIdHeader) {
-        if (StringUtils.isNotBlank(channelIdHeader)) {
-            this.credentialsFactory.isAuthenticationDisabled().thenCompose(isAuthDisabled -> {
-                if (isAuthDisabled) {
-                    return jwtTokenValidationValidateAuthHeader(authHeader, channelIdHeader, null)
-                                                                .thenCompose(claimsIdentity -> {
-                            String outboundAudience = SkillValidation.isSkillClaim(claimsIdentity.claims())
-                                ? JwtTokenValidation.getAppIdFromClaims(claimsIdentity.claims())
-                                : this.toChannelFromBotOAuthScope;
-
-                            return generateCallerId(this.credentialsFactory, claimsIdentity, this.callerId)
-                                                    .thenCompose(resultCallerId -> {
+
+        return this.credentialsFactory.isAuthenticationDisabled().thenCompose(isAuthDisabled -> {
+
+            if (StringUtils.isBlank(channelIdHeader) && !isAuthDisabled) {
+                return Async.completeExceptionally(
+                    new AuthenticationException("channelId header required when authentication is enabled"));
+
+            } else {
+                return jwtTokenValidationValidateAuthHeader(authHeader, channelIdHeader, null)
+                    .thenCompose(claimsIdentity -> {
+                        String outboundAudience = SkillValidation.isSkillClaim(claimsIdentity.claims())
+                            ? JwtTokenValidation.getAppIdFromClaims(claimsIdentity.claims())
+                            : this.toChannelFromBotOAuthScope;
+
+                        return generateCallerId(this.credentialsFactory, claimsIdentity, this.callerId)
+                            .thenCompose(resultCallerId -> {
                                 AuthenticateRequestResult authenticateRequestResult = new AuthenticateRequestResult();
                                 authenticateRequestResult.setClaimsIdentity(claimsIdentity);
                                 authenticateRequestResult.setAudience(outboundAudience);
                                 authenticateRequestResult.setCallerId(resultCallerId);
 
                                 return CompletableFuture.completedFuture(authenticateRequestResult);
                             });
-                        }
-                    );
-                }
-                return null;
-            });
-        }
-        throw new AuthenticationException("channelId header required");
+                    });
+            }
+        });
```

Comment:
Can we remove the else statement given that if the condition is truthful the execution will will fail through the early return? As a side effect, this may make it more readable and match with the C# implementation

Reply:
If I remove the else I have to add an "artificial" null return through that path, and I believe the _actual_ thing to return is the process that starts in line 136

Reply:
Are you sure? I'm not getting an error:

![image](https://user-images.githubusercontent.com/58704965/128245914-03ea5dcc-df8b-4df6-b607-99ce97ca110f.png)

Reply:
excellent. Going with it then